oauth2client/contrib/gce.py

```diff
@@ -48,28 +45,67 @@
 """
 
 
-def _get_service_account_email(http_request=None):
-    """Get the GCE service account email from the current environment.
-
+def _get_metadata(http_request=None, path=None, recursive=True):
```

Comment:
So there are just 3:
* `_METADATA_ROOT + 'instance/service-accounts/<email>/?recursive=true'`
Which returns basically a `service_account_info` object of which we use `scopes` and `email`

* `_METADATA_ROOT + 'project/project-id'` which returns the project-id as a string and will only be used for blob signing (or if someone is curious for some reason)

* `_METADATA_ROOT + 'instance/service-accounts/<email>/token'` which returns the token, which is only the two fields we use.

All the recursive nonsense is basically just to bundle the `scopes` and `email` call together...